src/main.rs

```diff
@@ -2,164 +2,44 @@
-use std::io;
-use std::io::{Cursor, Read, Seek, SeekFrom};
 
-use anyhow::ensure;
+#[macro_use]
+extern crate log;
+
 use bytesize;
 use hex;
-
-use filecoin_proofs::constants::DefaultPieceHasher;
-use filecoin_proofs::fr32::write_padded;
-use filecoin_proofs::{
-    generate_piece_commitment, PaddedBytesAmount, SectorSize, UnpaddedBytesAmount,
-};
-use storage_proofs::fr32::Fr32Ary;
-use storage_proofs::hasher::{Domain, Hasher};
-use storage_proofs::pieces::generate_piece_commitment_bytes_from_source;
-use storage_proofs::util::NODE_SIZE;
-
-type VecStore<E> = merkletree::store::VecStore<E>;
-pub type MerkleTree<T, A> = merkletree::merkle::MerkleTree<T, A, VecStore<T>>;
-
-// use a file as an io::Reader but pad out extra length at the end with zeros up
-// to the padded size
-struct PadReader {
-    file: File,
-    fsize: usize,
-    padsize: usize,
-    pos: usize,
-}
-
-impl io::Read for PadReader {
-    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        /*
-        if self.pos == self.fsize {
-          print!("reached file size, now padding ...")
-        }
-        */
-        if self.pos >= self.fsize {
-            for i in 0..buf.len() {
-                buf[i] = 0;
-            }
-            let cs = cmp::min(self.padsize - self.pos, buf.len());
-            self.pos = self.pos + cs;
-            /*
-            if cs < buf.len() {
-              print!("done with file ...")
-            }
-            */
-            Ok(cs)
-        } else {
-            let cs = self.file.read(buf)?;
-            self.pos = self.pos + cs;
-            Ok(cs)
-        }
-    }
-}
-
-// logic partly copied from Lotus' PadReader which is also in go-fil-markets
-// figure out how big this piece will be when padded
-fn padded_size(size: u64) -> u64 {
-    let logv = 64 - size.leading_zeros();
-    let sect_size = (1 as u64) << logv;
-    let bound = u64::from(UnpaddedBytesAmount::from(SectorSize(sect_size)));
-    if size <= bound {
-        return bound;
-    }
-    return u64::from(UnpaddedBytesAmount::from(SectorSize(1 << (logv + 1))));
-}
-
-// from rust-fil-proofs/src/storage_proofs/pieces.rs
-fn local_generate_piece_commitment_bytes_from_source<H: Hasher>(
-    source: &mut dyn Read,
-    padded_piece_size: usize,
-) -> anyhow::Result<Fr32Ary> {
-    ensure!(padded_piece_size > 32, "piece is too small");
-    ensure!(padded_piece_size % 32 == 0, "piece is not valid size");
-
-    let mut buf = [0; NODE_SIZE];
-    use std::io::BufReader;
-
-    let mut reader = BufReader::new(source);
-
-    let parts = (padded_piece_size as f64 / NODE_SIZE as f64).ceil() as usize;
-
-    let tree = MerkleTree::<H::Domain, H::Function>::try_from_iter((0..parts).map(|_| {
-        reader.read_exact(&mut buf)?;
-        <H::Domain as Domain>::try_from_bytes(&buf)
-    }))?;
-
-    let mut comm_p_bytes = [0; NODE_SIZE];
-    let comm_p = tree.root();
-    comm_p.write_bytes(&mut comm_p_bytes)?;
-
-    Ok(comm_p_bytes)
-}
+extern crate simple_logger;
```

Comment:
Not needed can simply be removed.

Reply:
why do I still need these bits:

```
#[macro_use]
extern crate lambda_runtime as lambda;
#[macro_use]
extern crate serde_derive;
```
?
is `log` set up different such that `info!()` and friends are exported in some special way and the serde and lambda macros aren't?

Reply:
nevermind, figured it out:

```
use lambda_runtime::error::HandlerError;
use lambda_runtime::{lambda, Context};
use serde_derive::{Deserialize, Serialize};
```